### PR TITLE
ci: make deploy dev pipelines

### PR DESCRIPTION
**Pull request by doty-pivotal**
Several make file clean up commits, the most important is:

make: deploy-pipeline is git aware

Previously the make target 'deploy-pipeline' was not savy to the
workflow differences of iterating on a branch as compared to setting
master. This commit removes this friction.  Now `make deploy-pipeline`
will try and do the right thing based on your git status.  If you are
checked out on master and tracking the production fork of the repo, it
should set the production version of the pipeline.  If not it will
interact with the dev deployment of our CI system.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,24 +1,14 @@
...
-HUB_PACKAGE=github.com/greenplum-db/gpupgrade/hub
 
 GIT_VERSION := $(shell git describe --tags | perl -pe 's/(.*)-([0-9]*)-(g[0-9a-f]*)/\1+dev.\2.\3/')
 UPGRADE_VERSION_STR="-X $(MODULE_NAME)/cli/commanders.UpgradeVersion=$(GIT_VERSION)"
 
-BRANCH := $(shell git for-each-ref --format='%(objectname) %(refname:short)' refs/heads | awk "/^$$(git rev-parse HEAD)/ {print \$$2}")
-LINUX_PREFIX := env GOOS=linux GOARCH=amd64
-MAC_PREFIX := env GOOS=darwin GOARCH=amd64
-LINUX_POSTFIX := .linux.$(BRANCH)
-MAC_POSTFIX := .darwin.$(BRANCH)
```
Comment (jchampio) [inline Makefile]: These are used below:
```make
 PREFIX = $($(OS)_PREFIX)
 POSTFIX = $($(OS)_POSTFIX)
```
Unfortunately you can't just grep for a variable name to see if it's used in Make; variable names themselves can be composed of other variables.

File: Makefile [hunk 2]
```diff
@@ -133,12 +133,15 @@ SECRETS_FILE ?= gpupgrade.dev.yml
...
-deploy-pipeline:
+set-pipeline:
 	fly -t $(FLY_TARGET) set-pipeline -p $(PIPELINE_NAME) \
 		-c ci/pipeline.yml \
 		-l ~/workspace/continuous-integration/secrets/$(SECRETS_FILE) \
 		-v gpupgrade-git-remote=$(GIT_URI) \
 		-v gpupgrade-git-branch=$(BRANCH) 
 
+destroy-pipeline:
+	fly -t $(FLY_TARGET) destroy-pipeline -p $(PIPELINE_NAME)
```
Comment (jchampio) [inline Makefile]: Do we do this enough to justify making it a top-level target? Combined with your new "auto-target master and prod" feature, I'm worried someone is going to accidentally blow away the production pipeline.

File: Makefile (hunk 1 above)
Comment (doty-pivotal) [reply Makefile]: Doh... I will put those back.

Comment (doty-pivotal) [reply Makefile]: Replaced

File: Makefile (hunk 2 above)
Comment (doty-pivotal) [reply Makefile]: Dropped.

**Comment by doty-pivotal** [comment]
Rebased on master.

**Pull request merged**